src/Daemon/Daemon.cpp

```diff
@@ -119,16 +119,53 @@ namespace usbguard
     _config.close();
   }
 
-  void Daemon::loadConfiguration(const std::string& path)
+  int Daemon::checkPermissions(const std::string& path,
+                               const mode_t permissions)
+  {
+    struct stat file_stat;
+    mode_t permission_bad { ( S_IRWXU | S_IRWXG | S_IRWXO ) - permissions };
```

Comment:
Please explain in a comment in the code why we need this set of hard coded permission flags and the logic of the check.